pkg/ovs/ovsctl/ovsctl.go

```diff
@@ -406,6 +406,46 @@ func (c *ovsCtlClient) SetPortNoFlood(ofport int) error {
 	return nil
 }
 
+func (c *ovsCtlClient) GetMeterStats() (map[int]int, error) {
+	output, err := runOfctlCmd(true, "meter-stats", c.bridge)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get meter stats on bridge %s: %v", c.bridge, err)
+	}
+	return c.parseMeterOutput(output)
+}
+
+// ovs-ofctl -OOpenFlow15 meter-stats br-int
+// The output for this command is like below:
+// OFPST_METER reply (OF1.5) (xid=0x2):
+// meter:1 flow_count:2 packet_in_count:100504 byte_in_count:14271500 duration:757.718s bands:
+// 0: packet_count:99849 byte_count:14178558
+// meter:2 flow_count:0 packet_in_count:0 byte_in_count:0 duration:1150.766s bands:
+// 0: packet_count:0 byte_count:0
+func (c *ovsCtlClient) parseMeterOutput(meterOutput []byte) (map[int]int, error) {
+	scanner := bufio.NewScanner(strings.NewReader(string(meterOutput)))
+	scanner.Split(bufio.ScanLines)
+	meterMap := make(map[int]int)
+	meterId := 1
+	for scanner.Scan() {
+		meterStr := scanner.Text()
+		if strings.HasPrefix(meterStr, "0:") {
+			strs := strings.Split(meterStr, " ")
+			for _, s := range strs {
+				pair := strings.Split(s, ":")
+				if pair[0] == "packet_count" {
+					val, err := strconv.Atoi(pair[1])
+					if err != nil {
+						return nil, err
+					}
+					meterMap[meterId] = val
+					meterId++
+				}
+			}
+		}
```

Comment:
Could you check if possible to use regex?

Reply:
Will remove this function.